app/models/thread.php

```diff
@@ -0,0 +1,88 @@
+<?php
+class Thread extends AppModel                    
+{
+    public $validation = array('title'=>array('length'=>array('validate_between',1,30),),); 
+
+    public static function getAll($offset, $limit)                
+    {
+        $threads = array();
+        $db = DB::conn();
+        $rows = $db->rows("SELECT * FROM thread LIMIT {$offset}, {$limit}");
+        
+        foreach ($rows as $row) {                    
+            $threads[] = new Thread($row);
+        }
+        
+        return $threads;
+    }   
+
+    public function create(Comment $comment)
+    {
+        $this->validate();
+        $comment->validate();
```

Comment:
Please separate thread and comment functions, Thread object/model is commonly only for thread table.
and Comment object/models is for comment table.